### ISSUE TITLE
Negative used value on smbfsm osx 10.9.1

### ISSUE THREAD

**Issue by giampaolo** [giampaolo/psutil#484]
_From [Pre...@gmail.com](https://code.google.com/u/102222347574037988465/) on February 24, 2014 23:12:54_
    
    Disk usage on my nas (smb) reports used diskspace as negative value:
    
         usage(total=616329322496, used=-3369346490368, free=3985675812864, 
    percent=-546.7)
        partition(device='//GUEST:@Tower._smb._tcp.local/AUTODOWNLOAD', 
    mountpoint='/Volumes/AUTODOWNLOAD', fstype='smbfs', 
    opts='rw,nosuid,ignore-ownership')
    
_Original issue: http://code.google.com/p/psutil/issues/detail?id=484_

**Comment by giampaolo** [comment]
I will assume this is an OSX bug and close this issue.

**Comment by albertvaka** [comment]
Hi @giampaolo. We found that `df` manages to return the size correctly even on MacOS (which, indeed, uses a type too small to hold sizes > 2Tb). I've checked the code for `df` and I found that they do some bitwise manipulation [1] and expand the number to a bigger type. I'm assuming this is what fixes the problem for them. Is this something that psutils could aim to do?

https://github.com/coreutils/gnulib/blob/master/lib/fsusage.c#L130
